crates/compute/src/memory.rs

```diff
@@ -71,10 +71,10 @@ pub trait ComputeMemory<F> {
 	const ALIGNMENT: usize;
 
 	/// An opaque handle to an immutable slice of elements stored in a compute memory.
-	type FSlice<'a>: Copy + SizedSlice + Sync;
+	type FSlice<'a>: Copy + SizedSlice + Sync + Debug;
```

Comment:
I'm not sure we need to derive Debug here, since it's unclear what that would mean for non-CPU implementations.
Maybe @jimpo, @GraDKh, or @BenjaminTrapani have an answer.